components/script/dom/document.rs

```diff
@@ -2014,6 +2018,67 @@ impl Document {
                            ReflowQueryType::NoQuery,
                            ReflowReason::ElementStateChanged);
     }
+
+    #[allow(unrooted_must_root)]
+    pub fn enter_fullscreen(&self, node: &Element) -> Rc<Promise> {
+        // Step 1
+        let promise = Promise::new(self.global().r());
+
+        self.fullscreen_element.set(Some(node));
+        if let Some(ref element) = self.fullscreen_element.get() {
+            // Step 4
+            // fullscreen element ready check
+            if !element.is_connected() {
+                // Step 7.1
+                self.upcast::<EventTarget>().fire_simple_event("fullscreenerror");
+                promise.reject(self.global().r().get_cx(), HandleValue::undefined());
+                return promise
+            }
+
+            // Step 6
+            let event = ConstellationMsg::SetFullscreenState(true);
+            self.window.upcast::<GlobalScope>().constellation_chan().send(event).unwrap();
+            element.set_fullscren_state(true);
```

Comment:
We will either need to disallow calling fullscreen on elements that are inside of iframes, or implement steps 7.2-7.5 as well. I'm fine with filing a followup issue for that as long as we disallow iframe support completely.